src/util/usertools.c

```diff
@@ -570,12 +571,32 @@ sss_fqname(char *str, size_t size, struct sss_names_ctx *nctx,
                               name, domain->name, calc_flat_name (domain), NULL);
 }
 
-errno_t sss_user_by_name_or_uid(const char *input, uid_t *_uid, gid_t *_gid)
+errno_t sss_user_by_name_or_uid(struct sss_nss_ops *ops, const char *input,
```

Comment:
Did you consider making `sss_nss_ops *ops` a static variable in this function?

Is there any reason you need to propogate it thru all interfaces? IIUC, it's opaque / not used anywhere besides this function.

Reply:
I did. Please take a look at the latest version of the changes.